provider/inmemory.go

```diff
@@ -78,13 +68,30 @@ func NewInMemoryProviderWithDomainAndLogging(domain string) *InMemoryProvider {
 			for _, v := range changes.Delete {
 				log.Infof("DELETE: %v", v)
 			}
-		},
-		OnRecords: func() {},
-		domain:    domain,
-		client:    newInMemoryClient(),
+		}
```

Comment:
Given there are multiple options (in the future) that would require to hook in via the `OnApplyChanges` func, then they would need to wrap any exiting value or something similar?